### ISSUE TITLE
XCP 8.3 coretemp module doesn't detect any sensors

### ISSUE THREAD

**Issue by mikhail-shevtsov-wiregate** [xcp-ng/xcp#669]

**Comment by olivierlambert** [comment]
Pinging @gduperrey

**Comment by SlashQuit** [comment]
Same/similar as per;
https://github.com/xcp-ng/xcp/issues/415
https://github.com/xcp-ng/xcp/issues/249
... and then one in the "Feedback" section at;
https://xcp-ng.org/forum/topic/9765/coretemp-alt-no-longer-works-with-8-3

This includes with a rebuilt coretemp.ko from; 
https://github.com/rushikeshjadhav/coretemp
(Using 8.3 instead of 8.0, and a 8.3 xcp-ng-build-env from ...)
https://github.com/xcp-ng/xcp-ng-build-env

**Comment by wechka53** [comment]
Same here. No sensors found from release 8.3. 
8.2 and even 8.3 betas were working fine

**Comment by rwjack** [comment]
Sooo... We were okay with no per thread temp readings, but no temp readings at all is straight up wack.

**Comment by stormi** [comment]
In the short term, would reading from IPMI be an acceptable workaround?

**Comment by rwjack** [comment]
damn, no love for homelabbers and consumer hardware. Anyways, I do have a single dell 5820 available, unsure if it has IPMI though.

**Comment by olivierlambert** [comment]
It's not a question of preferring some users vs others, it's really a Xen security/isolation problem in the first place. A solution is to not requesting the CPU capability directly from the Dom0 but going via an IPMI. I hope you understand we cannot leave a big security hole wide open and we had to fix it.

**Comment by rwjack** [comment]
Of course, I absolutely understand what you're saying. I'm a security guy myself, but on the other hand, I won't have much to secure if my servers die due to me not being able to monitor their health status (temps) ((yes, it's far fetched, but monitoring the temps gives me, and I think others, peace of mind)).

Do you think there any other potential solutions to this problem, other than buying enterprise level hardware? Perhaps getting an external pluggable device that could monitor the temps on consumer hardware? (If such a thing even exists)

Not sure if, for example, JetKVM could do the job? In theory it acts as an IPMI (kvm over IP + power control), but I'm doubtful it can read the temps as well.

This issue is literally the only thing barring me from jumping from 8.2.1, and testing 8.3.

**Comment by mikhail-shevtsov-wiregate** [comment]
@olivierlambert Just an idea. Maybe this security feature could be turned off during boot time with some boot flag?
Or it is impossible to do that due to `hardcode` in XEN hypervisor source?

**Comment by stormi** [comment]
That's something that is to be improved in Xen itself, to provide a way to access safely such data from the control domain which currently can't access that.

**Comment by rwjack** [comment]
So now that 8.3 is the LTS release, and 8.2.1 will be deprecated soon, is there any way at all to monitor cpu temps on non-enterprise hardware?

**Comment by olivierlambert** [comment]
I'm not aware of a trivial way to do it. I will make sure it's in the next XCP-ng team discussion so we have an idea of what can be done and where to put it in our backlog.

**Comment by olivierlambert** [comment]
Okay I asked internally, and it was already in the pipes for studying our different possibilities. So we'll come back here in few weeks (months tops) with the output of our investigation.

**Comment by bugsyb** [comment]
Amazing! Thank you @olivierlambert  - there are more of us, using xcp-ng at home and spreading out the word to where we have influence to help XCP-ng take over VMware deployments.

**Comment by olivierlambert** [comment]
I have no doubt about it, thank you very much! 👍  We'll try out best to find a way, stay tuned!

**Comment by TSnake41** [comment]
I'm investigating on a alternative approach for fetching the CPU temperature on Intel platforms; because Linux's coretemp cannot work properly when virtualized (well, it somewhat could with vCPU pinning, but it still had several flaws).

It is not exposed through Linux hwmon (thus not with lm-sensors); but would still be easy to fetch.

Here is some quick demo (on i5-4670).
```
# ./xen-intel-cputemp 
Package: 30°C

CPU0: 30°C
CPU1: 29°C
CPU2: 27°C
CPU3: 25°C
```

**Comment by rwjack** [comment]
@TSnake41 cool, can we test this on 8.3?

**Comment by TSnake41** [comment]
@rwjack 
> [@TSnake41](https://github.com/TSnake41) cool, can we test this on 8.3?

there is some progress in that area upstream : https://lore.kernel.org/xen-devel/cover.1761752801.git.teddy.astie@vates.tech/#t

**Comment by TSnake41** [comment]
See https://github.com/xcp-ng-rpms/xen/pull/78

**Comment by rwjack** [comment]
Thanks @TSnake41, does this mean this is ready for testing, how can we install it?

**Comment by TSnake41** [comment]
> Thanks [@TSnake41](https://github.com/TSnake41), does this mean this is ready for testing, how can we install it?

Not yet, it should land in the next (or the following one) testing cycle.

**Comment by rwjack** [comment]
Can you clarify this a bit for us end users? Are testing cycles an internal thing?

**Comment by TSnake41** [comment]
> Can you clarify this a bit for us end users? Are testing cycles an internal thing?

In https://xcp-ng.org/forum/topic/9964/xcp-ng-8-3-updates-announcements-and-testing, there are regular batches of updates (and testing period, which is what I call here a testing cycle).

This change should appear in a future update (very likely the next one)

**Comment by rwjack** [comment]
Understood, ty!

**Comment by rzr** [comment]
Hi we have some updates to share on this (old) request:

-  xen: Add support for xenpm get-core-temp to query CPU temperature on Intel platforms.
   -  Use xenpm get-core-temp to get the temperature on Intel's CPU, to fallback unsupported coretemp. Doc update being reviewed .

Free free to test our later packages from testing repository as explained at:

https://xcp-ng.org/forum/topic/9964/xcp-ng-8-3-updates-announcements-and-testing/584
